lib/src/rpc/dial.dart

```diff
@@ -0,0 +1,351 @@
+  /// AuthenticateTo() request to retrieve the credentials at the external auth endpoint.
+  ///
+  /// If enabled, other auth options have no affect. Eg. [signalingAuthEntity], [signalingCredentials],
+  /// [signalingExternalAuthAddress], [signalingExternalAuthToEntity]
+  String? signalingAccessToken;
+}
+
+Future<ClientChannelBase> dial(String address, DialOptions? options) async {
+  final opts = options ?? DialOptions();
+  _logger.i('Connecting to Robot at $address');
+  if (opts.webRtcOptions?.disable ?? false) {
+    return _dialDirectGrpc(address, opts);
+  }
+  return _dialWebRtc(address, opts);
+}
+
+Future<ClientChannelBase> _dialDirectGrpc(String address, DialOptions options) async {
+  _logger.d('Dialing direct GRPC to $address');
+  return _authenticatedChannel(address, options);
+}
+
+Future<ClientChannelBase> _dialWebRtc(String address, DialOptions options) async {
+  _logger.d('Dialing WebRTC to $address');
+  if (options.authEntity.isNullOrEmpty) {
+    if (options.externalAuthAddress.isNullOrEmpty) {
+      _logger.d('Auth entity is empty, setting to host: $address');
+      options.authEntity = address;
+    } else {
+      _logger.d('Auth entity is empty, setting to external auth address: ${options.externalAuthAddress}');
+      options.authEntity = options.externalAuthAddress;
+    }
+  }
+
+  final signalingServer = options.webRtcOptions?.signalingServerAddress ?? 'app.viam.com';
+  _logger.d('Connecting to signaling server: $signalingServer');
+  final signalingChannel = await _dialDirectGrpc(signalingServer, options);
+  _logger.d('Connected to signaling server: $signalingServer');
+  final signalingClient = SignalingServiceClient(signalingChannel, options: CallOptions(metadata: {"rpc-host": address}));
+  final config = (await signalingClient.optionalWebRTCConfig(OptionalWebRTCConfigRequest())).config;
+  final iceServers = config.additionalIceServers
+      .map((e) => {
+            'urls': e.urls,
+            'credential': e.credential,
+            'username': e.username,
+          })
+      .toList()
+    ..add({
+      'urls': 'stun:global.stun.twilio.com:3478?transport=udp',
+      'sdpSemantics': 'unified-plan',
+    });
+
+  final peerConnection = await createPeerConnection({'iceServers': iceServers});
+  final dataChannel = await peerConnection.createDataChannel(
+    'data',
+    RTCDataChannelInit()
+      ..binaryType = 'arraybuffer'
+      ..id = 0
+      ..negotiated = true
+      ..ordered = true,
+  );
+
+  final negotiationChannel = await peerConnection.createDataChannel(
+    'negotiation',
+    RTCDataChannelInit()
+      ..binaryType = 'arraybuffer'
+      ..id = 1
+      ..negotiated = true
+      ..ordered = true,
+  );
+
+  String? uuid;
+  final didConnect = Completer();
+  final didSetRemoteDesc = Completer();
+
+  // If tickleICE is enabled, set onIceCandidate handler
+  if (!(options.webRtcOptions?.disableTrickleIce ?? config.disableTrickle)) {
```

Comment:
`config.disableTrickle` was received by the server as an optional webrtc configuration. It shouldn't override `webrtcoptions.disable...` but it can be used as a backup in case the value the user passed in is empty